torch/nested/_internal/ops.py

```diff
@@ -79,6 +112,37 @@ def raggedness_matches(nt, size):
     return list(nt._size[:end]) == list(size[:end])
 
 
+def squeeze_leading_ones_get_extra(t, nt):
+    # Note: [ Squeezing leading ones ]
+    #
+    # Squeeze leading ones from t, and return the number of extra ones
+    # that need to be unsqueezed after relying on dense broadcasting
+    # Today, we would error if extra != 0 because NT does not yet support
+    # unsqueezing on the 0th dimension.
+    #
+    # We want:
+    #   (B, j0, ?, ?) + (1, 1, ?, ?) -> (B, j0, ?, ?)
+    #   (B, j0, ?, ?) + (1, 1, 1, ?, ?) -> (1, B, j0, ?, ?)  (not yet supported)
+    #
+    # 1) Squeeze extra ones and grab values from NT
+    #   (1, 1, ?, ?) -> (?, ?)   and   (sum(*), ?, ?) -> (B, j0, ?, ?)
+    # 2) Do dense broadcasting:
+    #   (sum(*), ?, ?) + (?, ?) -> (sum(*), ?, ?)
+    # 3) Construct nested tensor
+    #   (sum(*), ?, ?) -> (B, j0, ?, ?)
+    #
+    # If unsqueezing on the 0th dim becomes supported, we would unsqueeze
+    # at step (4).
+    t_shape = t.shape
+    extra = 0
+    for s in t_shape:
+        if s == 1:
+            if t.dim() > nt.dim():
+                extra += 1
+            t = t.squeeze(0)
```

Comment:
I'm reading this and it's hard to tell if it's correct. Looks like it depends on `t.squeeze(0)` being a no-op when `t.shape[0] > 1`. Also `1`s we see in the middle of t's shape shouldn't contribute to `extra`, right? Is it possible to simplify this and still get the correct guard behavior? For example:

```python
num_leading_ones = 0
while t.shape[0] == 1:
    num_leading_ones += 1
    t = t.squeeze(0)
return t, num_leading_ones
```

Reply:
Thanks, good catch, was missing the else-break here, and this simplification makes things much cleaner!